### PR TITLE
Implemented Radix Sort

### LINKED ISSUE #15: Implement the Radix Sort algorithm
https://github.com/sentenza/hacktoberfest-scala-algorithms/blob/1e541a09eb9d7eb0dbec7feebb3d1d13d7f4c06a/src/main/scala/io/github/sentenza/hacktoberfest18/sort/CommonSortings.scala#L34

> Radix sort is a non-comparative integer sorting algorithm that sorts data with integer keys by grouping keys by the individual digits which share the same significant position and value. A positional notation is required, but because integers can represent strings of characters (e.g., names or dates) and specially formatted floating point numbers, radix sort is not limited to integers.

### Contributing Guidelines
- Add the missing implementation of the [Radix Sort](https://en.wikipedia.org/wiki/Radix_sort)
- Add a test case in the spec that targets the `CommonSorting` object
- Don't be afraid to submit a new PR
- Have fun!

### PR DESCRIPTION
**Pull request by Xoeseko**
Resolved #15 added a parameter for the sorting basis as always feedback is greatly appreciated

### CODE REVIEW

**Comment by sentenza** [comment]
Hmmm, there's already this proposed solution under active review: #24

**Comment by Xoeseko** [comment]
Oops my bad... how did I not see that...

**Comment by sentenza** [comment]
> Hmmm, there's already this proposed solution under active review: #24

That one is still pending. I haven't received any feedback, so I reopened this one.

**Comment by Xoeseko** [comment]
@sentenza I refactored the code a little so that my old pull request is mergeable with the current state of things.. But I also saw the other contributor from #24 is back of course I would love feedback on my PR but I would gladly leave someone else a chance if need be..

File: src/main/scala/io/github/sentenza/hacktoberfest18/MenuIO.scala [hunk 1]
```diff
@@ -76,5 +76,4 @@ object MenuIO {
           renderInteractiveMenu()
       }
     }
-  }
```
Comment (sentenza) [inline src/main/scala/io/github/sentenza/hacktoberfest18/MenuIO.scala]: Why did you remove this bracket here?

File: src/main/scala/io/github/sentenza/hacktoberfest18/algos/MutableSorting.scala [hunk 2]
```diff
@@ -280,8 +280,53 @@ object MutableSorting extends Sorting[Array, Int] {
...
+        }
+        for (n <- arr){
+          // Start by getting the current digit and assigning the number to corresponding bucket.
+          val digit = (n / math.pow(b, it).toInt) % b
+          buckets(digit) += n
+        }
+      buckets.map(a => a.toArray)
+      }
 
+      def bucketsToList(buckets: Array[Array[Int]]): Array[Int] ={
```
Comment (sentenza) [inline src/main/scala/io/github/sentenza/hacktoberfest18/algos/MutableSorting.scala]: Should it be declared `private`? Would it be better to nest this method inside the `radixSort()` function?

File: src/main/scala/io/github/sentenza/hacktoberfest18/algos/MutableSorting.scala [hunk 3]
```diff
@@ -280,8 +280,53 @@ object MutableSorting extends Sorting[Array, Int] {
 
   }
 
-  def radixSort(xs: Array[Int]): Array[Int] = ???
+  /**
+    * @author Xoeseko
```
Comment (sentenza) [inline src/main/scala/io/github/sentenza/hacktoberfest18/algos/MutableSorting.scala]: The author tag must NOT be the first line in the scaladoc.

**Comment by sentenza** [comment]
> @sentenza I refactored the code a little so that my old pull request is mergeable with the current state of things.. But I also saw the other contributor from #24 is back of course I would love feedback on my PR but I would gladly leave someone else a chance if need be..

After days without any feedback this PR of yours got the right to be merged. ;)

File: src/main/scala/io/github/sentenza/hacktoberfest18/MenuIO.scala (hunk 1 above)
Comment (Xoeseko) [reply src/main/scala/io/github/sentenza/hacktoberfest18/MenuIO.scala]: I don't understand what this bracket and the next one close. To me the third bracket closes the MenuIO object and my IDE gives me an error when I include it I just tried checking out your branch to make sure the mistake wasn't coming from my code but it seems to be the same.

File: src/main/scala/io/github/sentenza/hacktoberfest18/algos/MutableSorting.scala (hunk 2 above)
Comment (Xoeseko) [reply src/main/scala/io/github/sentenza/hacktoberfest18/algos/MutableSorting.scala]: It is already embedded within the radixsort function isn't it ? Do I need to also make it private ? Why then if it is already encapsulated..?

File: src/main/scala/io/github/sentenza/hacktoberfest18/algos/MutableSorting.scala (hunk 3 above)
Comment (Xoeseko) [reply src/main/scala/io/github/sentenza/hacktoberfest18/algos/MutableSorting.scala]: Resolved

File: src/main/scala/io/github/sentenza/hacktoberfest18/algos/MutableSorting.scala (hunk 2 above)
Comment (sentenza) [reply src/main/scala/io/github/sentenza/hacktoberfest18/algos/MutableSorting.scala]: Sorry, my bad. ;)
